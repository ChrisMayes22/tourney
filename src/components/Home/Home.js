import React from 'react';
import classes from './Home.css';
import SubmitButton from '../SubmitButton/SubmitButton';

const home = (props) => {
    return(
        <div className={classes.background}>
            <nav className={classes.ui}>
                <h3>How many judges are joining us...?</h3>
                <input type="number"></input>
                <SubmitButton/>
            </nav>
        </div>
    )
}

export default home;