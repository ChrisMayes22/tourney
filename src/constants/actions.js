export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const REMOVE_LOWEST_SCORE = 'REMOVE_LOWEST_SCORE';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';
export const SET_PLAYERS = 'SET_PLAYERS';
export const RESET_FOR_FINALS = 'RESET_FOR_FINALS';
export const CHOOSE_WINNER = 'CHOOSE_WINNER';

export const initializeCharacter = function(imageUrl, id){
    return {type: INITIALIZE_CHARACTER, 
        payload: {image: imageUrl, id}}
}
export const removeCharacter = function(id){
    console.log('Remove Character Dispatched')
    console.log(REMOVE_CHARACTER)
    return {type: REMOVE_CHARACTER, id}
}
export const removeLowestScore = function(){
    return {type: REMOVE_LOWEST_SCORE};
}
export const rateCharacter = function(points, rowId, character){
    return {type: RATE_CHARACTER, payload:{
        points, rowId, character}}
}
export const submitCharacterRatings = function(playersArray){
    return {type: SUBMIT_CHARACTER_RATINGS, payload:{
        players: playersArray
    }};
};
export const setPlayers = function(num){
    return {type: SET_PLAYERS, number: num}
};
export const resetForFinals = function(){
    return {type: RESET_FOR_FINALS}
}
export const chooseWinner = function(character){
    return {type: CHOOSE_WINNER, character}
}

