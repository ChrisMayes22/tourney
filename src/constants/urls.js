export const VOTING_PAGE = '/voting';
export const ELIMINATION_PAGE = '/elimination';
export const NOT_ENOUGH_CHARACTERS = '/not-enough-characters';
export const UPLOAD_PAGE = '/upload';
export const WINNER_PAGE = '/winner';
export const HOME = '/';