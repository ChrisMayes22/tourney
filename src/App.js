import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import './App.css';
import Home from './components/Home/Home';
import CharacterRedirect from './components/CharacterRedirect/CharacterRedirect';
import VotingPage from './containers/VotingPage/VotingPage';
import UploadPage from './containers/UploadPage/UploadPage';

class App extends Component {
  render() {

    return (
      <Switch>
        <Route path='/voting-page' component={VotingPage}/>
        <Route path='/upload-page' component={UploadPage}/>
        <Route path='/not-enough-characters' component={CharacterRedirect}/>
        <Route path='/' component={Home}/>
      </Switch>
    );
  }
}

export default App;
