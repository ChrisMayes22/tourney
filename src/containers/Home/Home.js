import React, { Component } from 'react';
import classes from './Home.css';
import SubmitButton from '../../components/SubmitButton/SubmitButton';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'; 
import { setPlayers } from '../../constants/actions';
import * as urls from '../../constants/urls';

export class Home extends Component {

    render(){
        return(
            <div className={classes.background}>
                <nav className={classes.ui}>
                    <h3>How many judges are joining us...?</h3>
                    <input 
                        type="number"
                        onChange={(event) => this.props.onSetPlayers(Number(event.target.value))}></input>
                        {(this.props.players === 3 || this.props.players === 4)  ? 
                            <Link to={urls.UPLOAD_PAGE} id='uploadLink'>
                                <SubmitButton children={"Let's begin!"} id="beginButton"/>
                            </Link> : 
                            <h3>This game is designed for THREE or FOUR players</h3>}
                </nav>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        players: state.players
    }
};

const mapDispatchToProps = dispatch => {
    return {
        onSetPlayers: (num) => dispatch(setPlayers(num))
    }
 }

export default connect(mapStateToProps, mapDispatchToProps)(Home);