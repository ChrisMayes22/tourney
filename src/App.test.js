import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Home from './containers/Home/Home';
import componentNotFound from './components/ComponentNotFound/ComponentNotFound';
import App from './App';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import * as urls from './constants/urls';

//TODO: Either figure out a way to pass appropriate state to components that use large portions of the app's tree, OR
//Figure out how to test routes without invoking the entire tree of the component associated w/ the tested route.


describe('Basic render tests', function(){
  test('App renders as expected', () => {
    const wrapper = shallow(<App/>)
  
    expect(wrapper).toMatchSnapshot();
  })
})
describe('When rendering routes', function(){
  test('Invalid path should redirect to 404', () => {

    const initialState = {
      characters: [],
      players: 0
    }
  
    const mockReducer = function(state=initialState, action){
      return state
    }
  
    const store = createStore(mockReducer)
  
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[ '/rANdOmm' ]}>
          <App/>
        </MemoryRouter>
      </Provider>
    );
  
      expect(wrapper.find(Home)).toHaveLength(0);
      expect(wrapper.find(componentNotFound)).toHaveLength(1);
  });
  
  test('Valid path should not redirect to componentNotFound component.', () => {
    const initialState = {
      characters: [],
      players: 0
    }
  
    const mockReducer = function(state=initialState, action){
      return state
    }
  
    const store = createStore(mockReducer)
  
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[ urls.HOME ]}>
          <App/>
        </MemoryRouter>
      </Provider>
    );
  
    console.log(wrapper.debug())
  
      expect(wrapper.find(Home)).toHaveLength(1);
      expect(wrapper.find(componentNotFound)).toHaveLength(0);
  });
});