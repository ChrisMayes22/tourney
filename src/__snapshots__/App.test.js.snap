// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render tests App renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/voting"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/elimination"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/not-enough-characters"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/upload"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/winner"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/voting",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/elimination",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/not-enough-characters",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/upload",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/winner",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/voting"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/elimination"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/not-enough-characters"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/upload"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/winner"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/voting",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/elimination",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/not-enough-characters",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/upload",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/winner",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
